src/main/java/com/danielagapov/spawn/Services/User/UserService.java

```diff
@@ -396,24 +396,38 @@ public List<RecommendedFriendUserDTO> getRecommendedFriendsForUserId(UUID userId
             // Fetch the requesting user's friends
             List<UUID> requestingUserFriendIds = getFriendUserIdsByUserId(userId);
 
-            // Create a set of the requesting user's friends for quick lookup
-            Set<UUID> requestingUserFriendSet = new HashSet<>(requestingUserFriendIds);
+            // Fetch users who have already received a friend request from the user
+            List<UUID> sentFriendRequestIds = friendRequestService.getSentFriendRequestsByUserId(userId)
```

Comment:
```suggestion
            List<UUID> sentFriendRequestReceiverUserIds = friendRequestService.getSentFriendRequestsByUserId(userId)
```
Old name was misleading someone to read it as the friend request ids, not the receiver user ids of those friend requests